bayeux.go

```diff
@@ -234,7 +263,7 @@ func (b *Bayeux) connect(ctx context.Context, out chan MaybeMsg) chan MaybeMsg {
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
-		defer close(out)
+		defer status.disconnect(out)
```

Comment:
You have 3 deferred functions now, maybe you can replace it with a block `defer func() { ... }`?

Reply:
I will update it to:
`
		defer func() {
			close(out)
			wg.Done()
			status.disconnect()
		}()`